### ISSUE TITLE
[Feature request] Multiple text objects in one line without bullets or with align

### ISSUE THREAD

**Issue by vpetzel** [gitbrent/PptxGenJS#369]
The current system will not allow for multiple text objects without break if there is no bullet or some kind of align specified. This turns into a Problem if you consider, that text formatting (italic, bold, underlinde, strike, ...) are global to one text object. This means, that even code like this
```
<html>
  <head>
    <script lang="javascript" src="PptxGenJS/libs/jquery.min.js"></script>
    <script lang="javascript" src="PptxGenJS/libs/jszip.min.js"></script>
    <script lang="javascript" src="PptxGenJS/dist/pptxgen.js"></script>
  </head>
  <body>
    <script>
        pptx = new PptxGenJS();
        var slide = pptx.addNewSlide();

        slide.addText([
            { text: 'abc', options: {}},
            { text: 'def', options: {italic:true}}
        ], {x:'0', y:'0', w:'100%', h:'100%', align:'center', valign:'middle'});

//          Save PPTX
        pptx.save('Titel');
    </script>
  </body>
</html>
```
is not able to achieve this: abc_def_, but will output two seperate lines for abc and _def_.
This could be easily mended by adding an additional attribute to the text objects, let’s say, break, and change line 3668 from
```
else if ( idx > 0 && (typeof textObj.options.bullet !== 'undefined' || typeof textObj.options.align !== 'undefined') ) {
				strSlideXml += '</a:p><a:p>' + paragraphPropXml;
			}
```
to
```
else if ( idx > 0 && (typeof textObj.options.break === 'undefined' || textObj.options.break) && (typeof textObj.options.bullet !== 'undefined' || typeof textObj.options.align !== 'undefined') ) {
				strSlideXml += '</a:p><a:p>' + paragraphPropXml;
			}
```
and using the code
```
<html>
  <head>
    <script lang="javascript" src="PptxGenJS/libs/jquery.min.js"></script>
    <script lang="javascript" src="PptxGenJS/libs/jszip.min.js"></script>
    <script lang="javascript" src="PptxGenJS/dist/pptxgen.js"></script>
  </head>
  <body>
    <script>
        pptx = new PptxGenJS();
        var slide = pptx.addNewSlide();

        slide.addText([
            { text: 'abc', options: {}},
            { text: 'def', options: {italic:true, break:false}}
        ], {x:'0', y:'0', w:'100%', h:'100%', align:'center', valign:'middle'});

//          Save PPTX
        pptx.save('Titel');
    </script>
  </body>
</html>
```
which will achieve the desired effect.
See this patch:
[pptxgen.js.patch.txt](https://github.com/gitbrent/PptxGenJS/files/2148642/pptxgen.js.patch.txt)

**Comment by vpetzel** [comment]
abc_def_ is supposed to be abc and def in italic, but GitHub’s italic specified doesn’t seem to like prepending non-whitespace chars.

**Comment by bigianb** [comment]
+1 ... I've just come across this limitation.
There is already a breakline option. Maybe if that is explicitly set to false then the linebreak could be squashed.

**Comment by gitbrent** [comment]
I don't see where the library lacks the ability to format on a word-level basis.

This adds italics and bold without forcing a line-break or requiring bullets:
```
slide.addText(
  [
    { text:'1st word ', options:{ color:'99ABCC', italic:true } },
    { text:'2nd word ', options:{ color:'FFFF00', bold:true } },
    { text:'3rd word ', options:{ color:'0088CC' } }
  ],
  { x:1.5, y:1.5, w:6, h:2, margin:0.1, fill:'232323' }
);
```

![screen shot 2018-08-08 at 23 28 58](https://user-images.githubusercontent.com/7218970/43878119-16982f20-9b63-11e8-95f5-1d95f908eb95.png)

**Comment by vpetzel** [comment]
@gitbrent Your code does not use any kind of align, add align:'center' to the options and you end up with

![grafik](https://user-images.githubusercontent.com/40687043/43887790-17c97496-9bc0-11e8-9cf1-1b34770063cb.png)

**Comment by vpetzel** [comment]
Using the code
```
slide.addText(
    [
        { text:'1st word ', options:{ color:'99ABCC', italic:true } },
        { text:'2nd word ', options:{ color:'FFFF00', bold:true, break:false } },
        { text:'3rd word ', options:{ color:'0088CC', break: false } }
    ],
    { x:1.5, y:1.5, w:6, h:2, margin:0.1, fill:'232323', align:'center' }
    );
```
with the patch you get what you want
![grafik](https://user-images.githubusercontent.com/40687043/43887942-8a201a54-9bc0-11e8-95d6-8f48af847f5c.png)

**Comment by gitbrent** [comment]
But alignment necessitates a paragraph as you cant align words...

**Comment by vpetzel** [comment]
But you still have a paragraph, you just don’t add one for each text object. The problem is, that, as soon as an align is specified, you add one
`'</a:p><a:p>' + paragraphPropXml`
between the different text objects in the array.

**Comment by gitbrent** [comment]
Okay, so what problem are we trying to solve?

Do you have a screencap of the PptxGenJS output versus the result you're looking for.

**Comment by vpetzel** [comment]
When using the code
```
slide.addText(
    [
        { text:'1st word ', options:{ color:'99ABCC', italic:true } },
        { text:'2nd word ', options:{ color:'FFFF00', bold:true } },
        { text:'3rd word ', options:{ color:'0088CC' } }
    ],
    { x:1.5, y:1.5, w:6, h:2, margin:0.1, fill:'232323', align:'center' }
    );
```
I get 
![](https://user-images.githubusercontent.com/40687043/43887790-17c97496-9bc0-11e8-9cf1-1b34770063cb.png)

Note that the three text objects are not in the same line, so if I want to have it like this
![](https://user-images.githubusercontent.com/40687043/43887942-8a201a54-9bc0-11e8-95d6-8f48af847f5c.png)
there's no way to do it.

**Comment by PaddingtonTheBear** [comment]
@vpetzel, great job on writing this up and providing a fix in the mean time. I'm running into the same issue with this library when I want to style individual words in the PPT. 

I couldn't figure this out for the longest time but as you mentioned, it only occurs if you're using the 'align' options, which is a pretty common use case. Adding your patch works great. It should be included in the base library, or if the regular functionality doesn't want to be changed, then add a new config property for the "don't break" scenario.

**Comment by lejard-h** [comment]
Any news on this ? I am facing the same issue.

**Comment by gitbrent** [comment]
I'll get this patched in an upcoming release.

```
var pptx = new PptxGenJS();
pptx.layout = "LAYOUT_WIDE";
var slide = pptx.addSlide();

// Works
slide.addText(
  [
    { text: "1st line", options: { fontSize: 24, color: "99ABCC" } },
    { text: "2nd line", options: { fontSize: 36, color: "FFFF00" } },
    { text: "3rd line", options: { fontSize: 48, color: "0088CC" } }
  ],
  { x: 1.5, y: 1.5, w: 6, h: 2, margin: 0.1, fill: "232323" }
);

// Creates one word per line (bug: `align` is inherited/applied to each text object)
slide.addText(
  [
    { text: "1st word ", options: { color: "99ABCC", italic: true } },
    { text: "2nd word ", options: { color: "FFFF00", bold: true } },
    { text: "3rd word ", options: { color: "0088CC" } }
  ],
  { x: 1.5, y: 1.5, w: 6, h: 2, margin: 0.1, fill: "232323", align: "center" }
);

pptx.save("PptxGenJS-Issue#369");
```

<img width="652" alt="Screen Shot 2019-08-14 at 22 22 36" src="https://user-images.githubusercontent.com/7218970/63070812-4e79c400-bee2-11e9-9d53-7cf7e734bfc4.png">

**Comment by kupkama2** [comment]
hey @gitbrent, I just tried this in the 3.0.0-beta.4 and the problem is unfortunately still there.

**Comment by gitbrent** [comment]
Related: Issue #258

**Comment by gitbrent** [comment]
Fixed via #751
